src/Infrastructure/Adapter/BdTopoRoadGeocoder.php

```diff
@@ -177,4 +177,48 @@ public function computeReferencePoint(
             return Coordinates::fromLonLat($coordinates[0], $coordinates[1]);
         }
     }
+
+    public function findRoadNames(string $search, string $cityCode): array
+    {
+        // Build search query
+        // https://www.postgresql.org/docs/current/datatype-textsearch.html#DATATYPE-TSQUERY
+        $query = \str_replace(' ', ' & ', trim($search)) . ':*';
+
+        try {
+            $rows = $this->bdtopoConnection->fetchAllAssociative(
+                "
+                    SELECT array_to_string(
+                        array(
```

Comment:
(Détail) Peut-être ajouter un commentaire pour le "nous qui aura perdu le contexte"

```suggestion
                        -- BD TOPO contains lowercase road names. We capitalize non-stopwords.
                        -- Example: 'rue de france' -> 'Rue de France'. (Better than INITCAP('rue de france') -> 'Rue De France')
                        array(
                            SELECT CASE WHEN cardinality(t.lexemes) > 0 THEN INITCAP(t.token) ELSE t.token END
                            FROM ts_debug('french', nom_minuscule) AS t
                            WHERE t.alias NOT IN ('asciihword')
                        ),
```